kernel/kernel-api/src/main/java/io/delta/kernel/internal/types/DataTypeJsonSerDe.java

```diff
@@ -160,26 +165,31 @@ static DataType parseDataType(JsonNode json) {
    * Parses an <a href="https://github.com/delta-io/delta/blob/master/PROTOCOL.md#array-type">array
    * type </a>
    */
-  private static ArrayType parseArrayType(JsonNode json) {
+  private static ArrayType parseArrayType(
+      JsonNode json, String fieldPath, HashMap<String, String> collationMap) {
     checkArgument(
         json.isObject() && json.size() == 3,
         String.format("Expected JSON object with 3 fields for array data type but got:\n%s", json));
     boolean containsNull = getBooleanField(json, "containsNull");
-    DataType dataType = parseDataType(getNonNullField(json, "elementType"));
+    DataType dataType =
+        parseDataType(getNonNullField(json, "elementType"), fieldPath + "element", collationMap);
     return new ArrayType(dataType, containsNull);
   }
 
   /**
    * Parses an <a href="https://github.com/delta-io/delta/blob/master/PROTOCOL.md#map-type">map type
    * </a>
    */
-  private static MapType parseMapType(JsonNode json) {
+  private static MapType parseMapType(
+      JsonNode json, String fieldPath, HashMap<String, String> collationMap) {
     checkArgument(
         json.isObject() && json.size() == 4,
         String.format("Expected JSON object with 4 fields for map data type but got:\n%s", json));
     boolean valueContainsNull = getBooleanField(json, "valueContainsNull");
-    DataType keyType = parseDataType(getNonNullField(json, "keyType"));
-    DataType valueType = parseDataType(getNonNullField(json, "valueType"));
+    DataType keyType =
+        parseDataType(getNonNullField(json, "keyType"), fieldPath + "key", collationMap);
```

Comment:
there is no `.` before `.key`? Just like iceberg compat v2 nested field keys?

Reply:
it was bug, thanks
i didn't implement collationidentifier.equals, so that is reason why tests passed